src/coreclr/jit/gentree.cpp

```diff
@@ -6957,8 +6957,9 @@ GenTreeIntCon* Compiler::gtNewStringLiteralLength(GenTreeStrCon* node)
         return gtNewIconNode(0);
     }
 
-    int             length = -1;
-    const char16_t* str    = info.compCompHnd->getStringLiteral(node->gtScpHnd, node->gtSconCPX, &length);
+    constexpr int maxStrSize      = 1024;
+    char16_t      str[maxStrSize] = {};
+    int           length = info.compCompHnd->getStringLiteral(node->gtScpHnd, node->gtSconCPX, str, maxStrSize);
     if (length >= 0)
```

Comment:
Does `JITDUMP("Folded '\"%ws\".Length' to '%d'\n", str, length)` need the buffer to be zero-terminated? I do not see anything zero terminating the string when it does not fit into the buffer.

Reply:
Decided to remove %ws